libraries/networking/src/MessagesClient.h

```diff
@@ -49,19 +49,20 @@ class MessagesClient : public QObject, public Dependency {
     Q_OBJECT
 public:
     MessagesClient();
-    
+
     void startThread();
 
     /*@jsdoc
      * Sends a text message on a channel.
      * @function Messages.sendMessage
      * @param {string} channel - The channel to send the message on.
      * @param {string} message - The message to send.
-     * @param {boolean} [localOnly=false] - If <code>false</code> then the message is sent to all Interface, client entity, 
+     * @param {boolean} [localOnly=false] - If <code>false</code> then the message is sent to all Interface, client entity,
      *     server entity, and assignment client scripts in the domain.
-     *     <p>If <code>true</code> then: if sent from an Interface or client entity script it is received by all Interface and 
-     *     client entity scripts; if sent from a server entity script it is received by all entity server scripts; and if sent 
+     *     <p>If <code>true</code> then: if sent from an Interface or client entity script it is received by all Interface and
+     *     client entity scripts; if sent from a server entity script it is received by all entity server scripts; and if sent
      *     from an assignment client script it is received only by that same assignment client script.</p>
+     * @returns The size of sent data, or a negative error code.
```

Comment:
What are the error codes? These should be documented.

Reply:
I don't have a full grasp of all the code paths here to document it properly. It seems like we return -1 for various different kinds of preconditions, and eventually forward the return value of https://doc.qt.io/qt-5/qudpsocket.html#writeDatagram, which is also just -1 for different errors, that you can then check with the a different method. I wasn't 100% sure it's always just -1 so went with a generic description. I was tempted to dive in deeper here to identify all the error cases and define and return unique error codes, since that's the most convenient way to forward them to the C API, but didn't want to spend too much time on this at the current stage.

Reply:
Problem: `sendMessage()` will return 0 upon successfully sending the message if connected to a domain...
```
void MessagesClient::sendMessage(QString channel, QString message, bool localOnly) {
    ...
    auto packetList = encodeMessagesPacket(channel, message, senderID);
    return nodeList->sendPacketList(std::move(packetList), *messagesMixer);
    ...
}
```
`encodeMessagesPacket()` creates a reliable packet list:
```
auto packetList = NLPacketList::create(PacketType::MessagesData, QByteArray(), true, true);
```
This packet is sent by `sendPacketList()` using `writePacketList()`:
```
qint64 Socket::writePacketList(std::unique_ptr<PacketList> packetList, const SockAddr& sockAddr) {
    ...
    if (packetList->isReliable()) {
        ... queues the packet for sending and resending if necessary ...
        return 0;
    }
    ...
}
```
I.e., the message isn't sent synchronously so we can't tell whether it will succeed or fail.

I think the return values should be removed, reinstating the scripting API to how it was before.

Reply:
0 seems to me like a reasonable value to return to indicate that the message was scheduled, nothing was sent immediately but also no error occurred immediately. I can include that in the documentation, but otherwise I don't see a reason not to expose this information to the caller. 

In context of the C API I had a problem in the unit tests that this function was called before the message mixer was activated, so the message was just dropped, and the only way to diagnose it was to look in the logs. I thought to implement an explicit check for this specific case to return an error code, but then noticed that the send function already does that, along with some other pre-condition checks, so I just decided to expose/reuse that. If it's critical to not expose this in the scripting API, I'll create separate functions for use in the C API.

Reply:
MessagesClient.h, .cpp: `MessagesClient::sendMessage()`, `MessagesClient::sendData()`, and `MessagesClient::sendLocalMessage()` will only ever return:
- 0 if sending on the network.
- N=bytes if sending local-only.
I.e., never a negative number. 

In the context of the scripting API, a return value doesn't provide any information above what the script already knows (i.e., message size and whether sent locally).

If wanting to use a negative number to indicate error in the C API then the following code would be the equivalent of the current PR code:
```
    int Messages::send(std::bitset<8> type, QString channel, QByteArray payload, bool localOnly) {
        auto client = DependencyManager::get<MessagesClient>();

        if (type[MESSAGE_TYPE_TEXT_INDEX]) {
            client->sendMessage(channel, QString::fromUtf8(payload), localOnly);
            return localOnly ? QString::fromUtf8(payload).length : 0;
        }

        if (type[MESSAGE_TYPE_DATA_INDEX]) {
            client->sendData(channel, payload, localOnly);
            return localOnly ? QString::fromUtf8(payload).length : 0;
        }

        return -1;
    }
```

If really wanting to provide a return value from  `sendMessage()`, `sendData()`, and `sendLocalMessage()` then their documentation should say, "Returns the number of bytes sent if send local-only, otherwise 0."

Reply:
Documentation updated.